modules/core/include/opencv2/core/directx.hpp

```diff
@@ -45,6 +45,10 @@
 #include "mat.hpp"
 #include "ocl.hpp"
 
+#ifdef HAVE_OPENCL
+typedef struct _cl_mem * cl_mem;
```

Comment:
as far i rememner `cl_mem` is registered in OPENCL as reserved name already. we should no over-typedefed it

Reply:
Removed. Image2D\Context instead.